src/Common/Core/Impl/IO/IFileSystem.cs

```diff
@@ -28,5 +30,9 @@ public interface IFileSystem {
         void DeleteDirectory(string path, bool recursive);
         string[] GetFileSystemEntries(string path, string searchPattern, SearchOption options);
         void CreateDirectory(string path);
+
+        string CompressFile(string path);
+        string CompressDirectory(string path);
+        string CompressDirectory(string path, Matcher matcher, Action<string> callback);
```

Comment:
Also, perhaps a cancellation token (and check it before processing each entry in the implementation)? At some point we might want to do a fancier UI here, probably with a progress bar and a cancel button, and then it'll come in handy.

---
In reply to: [74853811](https://github.com/Microsoft/RTVS/pull/2171#discussion_r74853811) [](ancestors = 74853811)